### PR TITLE
Adding Student and Teacher tests to Oauth platforms

### PR DESCRIPTION
**Pull request by hannahbergam**
Adds student and teacher tests to match the existing new teacher test in the google oauth test file. 

This is an attempt to further test the logic added in the following three PRs:
1. [Oauth back end pr](https://github.com/code-dot-org/code-dot-org/pull/61496)
2. [Save student pr](https://github.com/code-dot-org/code-dot-org/pull/61616)
3. [Save teacher pr](https://github.com/code-dot-org/code-dot-org/pull/60761)

PR 1 above added one test for the teacher google flow. This PR adds tests for google and facebook for students (both) and teacher (facebook). To see the logic we are testing, I'd first check out PR #1. 

The existing clever testing doesn't touch the omniauth/redirect file I edited, but I am open to feedback about how to test that flow specifically. There also doesn't exist a microsoft test file. The flow for all four is nearly the same so I'm not sure if adding one would be redundant (and am unclear on the initial reasoning to not have one). Perhaps Platform can provide more context.

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2430

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by TurnerRiley — APPROVED** [review]
These look great! Wondering if we could have a Clever test that mostly checks the `SignUpTracking` events since then we'll know it's reaching all the points it should (such as `SignUpTracking.log_oauth_callback AuthenticationOption::CLEVER, session` in the `sign_up_clever` method)? I can only speak to the tests themselves though so I'll defer to Platform in terms of adding the Microsoft tests or not?

**Review by stephenliang — APPROVED** [review]

File: dashboard/test/integration/omniauth/facebook_test.rb [hunk 1]
```diff
@@ -15,6 +15,33 @@ class FacebookTest < ActionDispatch::IntegrationTest
       SignUpTracking.stubs(:split_test_percentage).returns(0)
     end
 
+    test "student sign up for newest sign up flow" do
+      auth_hash = mock_oauth
+
+      post "/users/auth/facebook"
+      get '/users/auth/facebook/callback', params: {finish_url: '/users/new_sign_up/finish_student_account'}
```
Comment (carl-codeorg) [inline dashboard/test/integration/omniauth/facebook_test.rb:22]: The utils file has some [sign_in_through](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/integration/omniauth/utils.rb#L40-L44) helpers that do the post to a provider and assert the redirect. Might dry this up a bit.

**Review by carl-codeorg — APPROVED** [review]
LGTM - just had one non-blocking suggestion.

File: dashboard/test/integration/omniauth/facebook_test.rb (hunk 1 above)
Comment (hannahbergam) [reply dashboard/test/integration/omniauth/facebook_test.rb:22]: I'm going to do some clean up in a followup and rework some of these tests to make sure they're truly testing the new flow. Merging this as is when staging reopens so I can start from this point!

**Pull request merged**